website_event_require_login/controllers/main.py

```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Tecnativa - David Vidal
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.http import request, route
+from odoo.addons.website_event.controllers.main import WebsiteEventController
+
+
+class RequireLoginToRegister(WebsiteEventController):
+
+    @route()
+    def registration_new(self, event, **post):
+        public_user = request.env.user == request.env.ref('base.public_user')
+        if public_user and event.website_require_login:
+            return request.env['ir.ui.view'].render_template(
```

Comment:
IMHO it would be better to just redirect the user to `"/web/login?redirect=%s" % event_url`. Don't you think so? We could save the need to maintain a separate template, and we'd save a click for the user.

Reply:
Sure the redirect is needed to keep the user on track, but the popup is necessary for the user to know that the event requires login. To kick him off without warning would be misleading I thing :)